### PR TITLE
Fixed #35270 -- Optimized model's Options._property_names.

### PR DESCRIPTION
**Pull request by adamchainz**
# Trac ticket number

ticket-35270

# Branch description

Covered in the ticket.

# Checklist
- [x] This PR targets the `main` branch. 
- [x] The commit message is written in past tense, mentions the ticket number, and ends with a period.
- [x] I have checked the "Has patch" **ticket flag** in the Trac system.
- [x] I have added or updated relevant **tests**.
- [x] I have added or updated relevant **docs**, including release notes if applicable.
- [n/a] For UI changes, I have attached **screenshots** in both light and dark modes.

### CODE REVIEW

**Comment by adamchainz** [comment]
Test failure looks like a flake.

**Review by felixxm — APPROVED** [review]
@adamchainz Thanks :+1:

File: django/db/models/options.py [hunk 1]
```diff
@@ -972,11 +971,15 @@ def total_unique_constraints(self):
...
-                names.append(name)
+        names = set()
+        for klass in self.model.__mro__:
+            names.update(
+                {
+                    name
+                    for name, value in klass.__dict__.items()
+                    if isinstance(value, property)
+                }
+            )
```
Comment (ngnpope) [inline django/db/models/options.py:982]: You can avoid all the call overhead for `.update()`:

```suggestion
            names |= {
                name
                for name, value in klass.__dict__.items()
                if isinstance(value, property)
            }
```

The code is cleaner and also slightly faster:

```pycon
In [1]: %%timeit x = set()
   ...: for i in range(100):
   ...:     x.update({"a", "b", "c"})
   ...: 
18.5 µs ± 2.77 µs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)

In [2]: %%timeit x = set()
   ...: for i in range(100):
   ...:     x |= {"a", "b", "c"}
   ...: 
13.1 µs ± 582 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```

**Review by ngnpope — APPROVED** [review]
Thanks 👍🏻

File: django/db/models/options.py (hunk 1 above)
Comment (adamchainz) [reply django/db/models/options.py:982]: I always forget the set operators, thanks! 👍

Comment (felixxm) [reply django/db/models/options.py:982]: It seems you forgot to commit Nick's suggestion.

Comment (adamchainz) [reply django/db/models/options.py]: I forgot to push it 🤦

**Pull request merged**